src/cmap/connection_pool_events.ts

```diff
@@ -102,19 +137,23 @@ export class ConnectionClosedEvent extends ConnectionPoolMonitoringEvent {
   /** The id of the connection */
   connectionId: number | '<monitor>';
   /** The reason the connection was closed */
-  reason: string;
+  reason: 'idle' | 'stale' | 'poolClosed' | 'error';
   serviceId?: ObjectId;
+  name = CONNECTION_CLOSED;
+  error: Error | undefined;
```

Comment:
```suggestion
  error: Error | null;
```

If we can using null is preferred to indicate the absence of something.
Also should this be MongoError? Following the logic it seems like the only non MongoError that could be set here is socket.authorizationError in connect.ts, maybe that's something we can add TODOs for but generally this type is more useful if it's a MongoError, b/c it will have a `.code` property

Reply:
The call to `connect` in connection_pool.ts where `ConnectionClosedEvent` is instantiated accepts a callback of the shape `(error?: AnyError, result?:Connection) => void`, so I think it would be appropriate to have this be `AnyError`

Reply:
`AnyError` is a vestige of the `Callback` type which wasn't fully accurate. So in reality  `connect` will always return a MongoError subclass, besides one case: `authorizationError` -- I'm wondering if we can keep the TypeScript on this class accurate to what we _desire_ (MongoError) before making this public, and we can follow up with wrapping that one outlier later. 

I believe, wrapping it in a MongoError would be acceptable in semver per our policy on errors descending from the same parent + I'm not sure if `authorizationError` would be surfaced to user's via any other mechanism or if it would be converted to a server selection timeout error.

Reply:
From IRL, let's make the type MongoError, cast away issues, file a follow up to wrap `authorizationError`, add todos.